src/Nri/Ui/Text/V4.elm

```diff
@@ -0,0 +1,193 @@
+
+@docs ugMediumBody, ugSmallBody
+
+
+## Modifying strings to display nicely:
+
+@docs noWidow
+
+-}
+
+import Css exposing (..)
+import Html.Styled exposing (..)
+import Html.Styled.Attributes exposing (css)
+import Nri.Ui.Colors.V1 exposing (..)
+import Nri.Ui.Fonts.V1 as Fonts
+
+
+{-| This is some medium body copy.
+-}
+mediumBody : List (Html msg) -> Html msg
+mediumBody content =
+    p
+        [ paragraphStyles
+            { font = Fonts.baseFont
+            , color = gray20
+            , size = 18
+            , lineHeight = 28
+            , weight = 400
+            , margin = 10
+            }
+        ]
+        content
+
+
+{-| This is some small body copy.
+-}
+smallBody : List (Html msg) -> Html msg
+smallBody content =
+    p
+        [ paragraphStyles
+            { font = Fonts.baseFont
+            , color = gray20
+            , size = 15
+            , lineHeight = 23
+            , weight = 400
+            , margin = 7
+            }
+        ]
+        content
+
+
+{-| This is some small body copy but it's gray.
+-}
+smallBodyGray : List (Html msg) -> Html msg
+smallBodyGray content =
+    p
+        [ paragraphStyles
+            { font = Fonts.baseFont
+            , color = gray45
+            , size = 15
+            , lineHeight = 23
+            , weight = 400
+            , margin = 7
+            }
+        ]
+        content
+
+
+paragraphStyles config =
+    css
+        [ config.font
+        , fontSize (px config.size)
+        , color config.color
+        , lineHeight (px config.lineHeight)
+        , fontWeight (int config.weight)
+        , padding zero
+        , textAlign left
+        , margin4 (px 0) (px 0) (px config.margin) (px 0)
+        , lastChild
+            [ margin zero
+            ]
+        ]
+
+
+{-| This is a little note or caption.
+-}
+caption : List (Html msg) -> Html msg
+caption content =
+    p
+        [ paragraphStyles
+            { font = Fonts.baseFont
+            , color = gray45
+            , size = 13
+            , lineHeight = 18
+            , weight = 400
+            , margin = 5
+            }
+        ]
+        content
+
+
+{-| User-generated text.
+-}
+ugMediumBody : List (Html msg) -> Html msg
+ugMediumBody =
+    p
+        [ css
+            [ Fonts.quizFont
+            , fontSize (px 18)
+            , lineHeight (px 30)
+            , whiteSpace preLine
+            , color gray20
+            , margin zero
+            ]
+        ]
+
+
+{-| User-generated text.
+-}
+ugSmallBody : List (Html msg) -> Html msg
+ugSmallBody =
+    p
+        [ css
+            [ Fonts.quizFont
+            , fontSize (px 16)
+            , lineHeight (px 25)
+            , whiteSpace preLine
+            , color gray20
+            , margin zero
+            ]
+        ]
+
+
+{-| Eliminate widows (single words on their own line caused by
+wrapping) by inserting a non-breaking space if there are at least two
+words.
+-}
+noWidow : String -> String
```

Comment:
Will headings not need this function?

Reply:
this just turns "one two three" into "one two&nbsp;three" to avoid widows. It's specific to text, although it is applicable to headers too.